packages/xdg_icons/lib/src/icon.dart

```diff
@@ -131,15 +132,33 @@ class XdgIconState extends State<XdgIcon> {
     final file = File(_icon?.fileName ?? '');
     final size = _resolveSize();
     if (file.existsSync()) {
-      final builder = _icon!.isScalable ? SvgPicture.file : Image.file;
-      return builder(
+      if (_icon!.isScalable) {
+        return ScalableImageWidget.fromSISource(
+          si: ScalableImageSource.fromSvgFile(
+            file,
+            file.readAsString,
+          ),
+        );
+      }
+
+      return Image.file(
         file,
         width: size.toDouble(),
         height: size.toDouble(),
       );
     } else if (_icon?.data != null) {
-      final builder = _icon!.isScalable ? SvgPicture.memory : Image.memory;
-      return builder(
+      if (_icon!.isScalable) {
+        final memoryFile = MemoryFileSystem().file('${_icon!.fileName}.svg');
+        memoryFile.writeAsBytesSync(_icon!.data!);
```

Comment:
Could one avoid the extra copy of the SVG resource data to the memory FS by returning/decoding `icon.data` directly from the second `fileReader` argument of `ScalableImageSource.fromSvgFile`? Notice that the first `Object file` argument doesn't need to be an actual `File` - it is only used for equality comparison.

Reply:
It seems you can. Nice, that avoids memory usage and a dependency! Pushed the change.